### PR TITLE
update code base so that it works with fo-dicom 4.0.2  (issue #33)

### PR DESCRIPTION
**Pull request by shelbypereira**
Most of the update is straightforward.  Many of the unit tests needed to be updated as the latest fo-dicom is stricter in its validation (a good thing!).    The changes fix all warning messages related to obsolete code.  This addresses the upgrade issue #33

### CODE REVIEW

**Comment by shelbypereira** [comment]
During my update of the code I noticed that my update to the files  JsonDicomConverter.cs and XmlDicomConverter.cs  (which address the obsolete AddOrUpdatePixelData warning) may be suspect.  By looking at my change:
FROM: dataset.AddOrUpdatePixelData ( vr, data, fo.DicomTransferSyntax.Parse ( TransferSyntaxUID ) ) ;

TO:
 var pixelData= DicomPixelData.Create(dataset, true);  //2nd parameter is true since we are adding new data here
                    pixelData.AddFrame(data);

you will see tha the vr parameter is not used in the new version.  In fact it is not clear how to set it.  I have asked the question on gitter here:
https://gitter.im/fo-dicom/fo-dicom#
Also note the comment I made on gitter:

I think this old thread: https://groups.google.com/forum/#!topic/fo-dicom/4j4OCwHXTJw may be relevant as people mention an issue with swapping of bytes. However there was no resolution and the last comment was unanswered. The comment in the fo-dicom lib for the original AddorUpdatePixelData read (for the vr param): DICOM vr of the image pixel. For a PixelData element this value should be either DicomVR.OB or DicomVR.OW DICOM VR. so short question is how is this functioning now that the param is not providing, and is there some other way to set this to be 100% sure to have the original behaviour.

That being said,  I think this is minor even if it opens up a bug (which would be more of a bug in fo-dicom).  

I also think we may need more unit tests, but I am not familiar enough with the library at this point.  Since the new fo-dicom is much stricter in its validation,  there may be cases which were not hit by the existing tests.  There were quite a few fixes needed to  get the existing tests to work although most were in the test code and not in the main code.  

Looking forward to any feedback!

**Comment by Zaid-Safadi** [comment]
@shelbypereira thank you for the great effort!

The changes look good to me and I like to review the points you mentioned.

I'll merge your changes into a new branch and we can test it there

**Comment by shelbypereira** [comment]
Concerning the transfer syntax,  the old method  took the TransferSyntax as well so I assume it is using it in a similar way.  with respect to the VR parameter,  I received an answer on the fo-dicom gitter:
from @gofal:
@shelbypereira When looking at the code in DicomPixelData.Create I assume the following: the dataset passed as the first parameter in DicomPixelData.Create method has to contain a value in DicomTag.BitsAllocated. If the value in BitsAllocated is greater than 8, then a VR of OW is assumed, else OB is used 

So in other words the new method assumes that the dataset has been setup appropriately.  I guess to test this rigorously we would need three different images with different values of BitsAllocated.  A true unit test might be difficult to write, but it would be useful if as a quick test we can somehow use 3 different images and then run the code and see if we can view the images correctly after processing.
